File: problems/problems_3408/Solution.cpp
//go:build ignore
#include "cpp/common/Solution.h"


using namespace std;
using json = nlohmann::json;

class TaskManager {
public:
    TaskManager(vector<vector<int>>& tasks) {
        
    }
    
    void add(int userId, int taskId, int priority) {
        
    }
    
    void edit(int taskId, int newPriority) {
        
    }
    
    void rmv(int taskId) {
        
    }
    
    int execTop() {
        
    }
};

/**
 * Your TaskManager object will be instantiated and called as such:
 * TaskManager* obj = new TaskManager(tasks);
 * obj->add(userId,taskId,priority);
 * obj->edit(taskId,newPriority);
 * obj->rmv(taskId);
 * int param_4 = obj->execTop();
 */

json leetcode::qubh::Solve(string input_json_values) {
	vector<string> inputArray;
	size_t pos = input_json_values.find('\n');
	while (pos != string::npos) {
		inputArray.push_back(input_json_values.substr(0, pos));
		input_json_values = input_json_values.substr(pos + 1);
		pos = input_json_values.find('\n');
	}
	inputArray.push_back(input_json_values);

	vector<string> operators = json::parse(inputArray[0]);
	vector<vector<json>> op_values = json::parse(inputArray[1]);
	auto obj0 = make_unique<TaskManager>(op_values[0][0]);
	vector<json> ans = {nullptr};
	for (size_t i = 1; i < op_values.size(); ++i) {
		if (operators[i] == "add") {
			obj0->add(op_values[i][0], op_values[i][1], op_values[i][2]);
			ans.push_back(nullptr);
			continue;
		}
		if (operators[i] == "edit") {
			obj0->edit(op_values[i][0], op_values[i][1]);
			ans.push_back(nullptr);
			continue;
		}
		if (operators[i] == "rmv") {
			obj0->rmv(op_values[i][0]);
			ans.push_back(nullptr);
			continue;
		}
		if (operators[i] == "execTop") {
			ans.push_back(obj0->execTop());
			continue;
		}
		ans.push_back(nullptr);
	}
	return ans;
}

File: problems/problems_3408/Solution.java
package problems.problems_3408;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


class TaskManager {

    public TaskManager(List<List<Integer>> tasks) {
        
    }
    
    public void add(int userId, int taskId, int priority) {
        
    }
    
    public void edit(int taskId, int newPriority) {
        
    }
    
    public void rmv(int taskId) {
        
    }
    
    public int execTop() {
        
    }
}

/**
 * Your TaskManager object will be instantiated and called as such:
 * TaskManager obj = new TaskManager(tasks);
 * obj.add(userId,taskId,priority);
 * obj.edit(taskId,newPriority);
 * obj.rmv(taskId);
 * int param_4 = obj.execTop();
 */

public class Solution extends BaseSolution {


    @Override
    public Object solve(String[] inputJsonValues) {
        String[] operators = jsonArrayToStringArray(inputJsonValues[0]);
		String[][] opValues = jsonArrayToString2DArray(inputJsonValues[1]);
		List<List<Integer>> tasks = jsonArrayTo2DIntList(opValues[0][0]);
		TaskManager obj = new TaskManager(tasks);
		List<Object> ans = new ArrayList<>(operators.length);
		ans.add(null);
		for (int i = 1; i < operators.length; i++) {
			if (operators[i].compareTo("add") == 0) {
				int userId = Integer.parseInt(opValues[i][0]);
				int taskId = Integer.parseInt(opValues[i][1]);
				int priority = Integer.parseInt(opValues[i][2]);
				obj.add(userId, taskId, priority);
				ans.add(null);
				continue;
			}
			if (operators[i].compareTo("edit") == 0) {
				int taskId = Integer.parseInt(opValues[i][0]);
				int newPriority = Integer.parseInt(opValues[i][1]);
				obj.edit(taskId, newPriority);
				ans.add(null);
				continue;
			}
			if (operators[i].compareTo("rmv") == 0) {
				int taskId = Integer.parseInt(opValues[i][0]);
				obj.rmv(taskId);
				ans.add(null);
				continue;
			}
			if (operators[i].compareTo("execTop") == 0) {
				
				ans.add(obj.execTop());
				continue;
			}
			ans.add(null);
		}
        return JSON.toJSON(ans);
    }
}
